collections/src/main/scala/strawman/collection/TupleZipped.scala

```diff
@@ -0,0 +1,347 @@
+    bf.fromSpecificIterable(coll1)(new View[B] {
+      def iterator() = new Iterator[B] {
+        private val elems1 = coll1.iterator()
+        private val elems2 = coll2.iterator()
+        def hasNext = elems1.hasNext && elems2.hasNext
+        def next() = f(elems1.next(), elems2.next())
+      }
+    })
+  }
+
+  def flatMap[B, C](f: (El1, El2) => Iterable[B])(implicit bf: BuildFrom[C1, B, C]): C = {
+    bf.fromSpecificIterable(coll1)(new View[B] {
+      def iterator() = new Iterator[B] {
+        private val elems1 = coll1.iterator()
+        private val elems2 = coll2.iterator()
+        private var _current: Iterator[B] = Iterator.empty
+        private def current = {
+          while (!_current.hasNext && elems1.hasNext && elems2.hasNext)
+            _current = f(elems1.next(), elems2.next()).iterator()
+          _current
+        }
+
+        def hasNext = current.hasNext
+        def next() = current.next()
+      }
+    })
+  }
+
+  def filter[C](p: (El1, El2) => Boolean)(implicit bf: BuildFrom[C1, (El1, El2), C]): C = {
+    bf.fromSpecificIterable(coll1)(new View[(El1, El2)] {
+      def iterator() = new Iterator[(El1, El2)] {
+        private val elems1 = coll1.iterator()
+        private val elems2 = coll2.iterator()
+        private var _current: (El1, El2) = _
+        private def current = {
+          while ((_current eq null) && elems1.hasNext && elems2.hasNext) {
+            val e1 = elems1.next()
+            val e2 = elems2.next()
+            if (p(e1, e2)) _current = (e1, e2)
+          }
+          _current
+        }
+        def hasNext = current ne null
+        def next() = {
+          val c = current
+          if (c ne null) {
+            _current = null
+            c
+          } else Iterator.empty.next()
+        }
+      }
+    })
+  }
+
+  def exists(p: (El1, El2) => Boolean): Boolean = {
+    val elems1 = coll1.iterator()
+    val elems2 = coll2.iterator()
+    var res = false
+
+    while (!res && elems1.hasNext && elems2.hasNext) res = p(elems1.next(), elems2.next())
+
+    res
+  }
+
+  def forall(p: (El1, El2) => Boolean): Boolean = !exists((el1, el2) => !p(el1, el2))
+
+  def foreach[U](f: (El1, El2) => U): Unit = {
+    val elems1 = coll1.iterator()
+    val elems2 = coll2.iterator()
+
+    while (elems1.hasNext && elems2.hasNext) f(elems1.next(), elems2.next())
+  }
+
+  private def iterator() = new Iterator[(El1, El2)] {
+    private val elems1 = coll1.iterator()
+    private val elems2 = coll2.iterator()
+    def hasNext = elems1.hasNext && elems2.hasNext
+    def next() = (elems1.next(), elems2.next())
+  }
+
+  def className = getClass.getName
+
+  override def toString = s"$className($coll1, $coll2)"
+}
+
+object Tuple2Zipped {
+  implicit def tuple2ZippedToIterable[El1, C1 <: Iterable[El1],
+                                      El2, C2 <: Iterable[El2]](zipped2: Tuple2Zipped[El1, C1, El2, C2]): Iterable[(El1, El2)] =
+    new View[(El1, El2)] { def iterator() = zipped2.iterator() }
```

Comment:
Maybe we should be more explicit and also put `View[(El1, El2)]` as the static return type?

Reply:
ok, will do